pkg/cmd/infra/deployer/deployer.go

```diff
@@ -196,5 +203,53 @@ func (d *Deployer) Deploy(namespace, deploymentName string) error {
 	} else {
 		glog.Infof("Deploying from %s to %s (replicas: %d)", deployutil.LabelForDeployment(from), deployutil.LabelForDeployment(to), desiredReplicas)
 	}
-	return strategy.Deploy(from, to, desiredReplicas)
+	if err := strategy.Deploy(from, to, desiredReplicas); err != nil {
+		return err
+	}
+	return d.after(config, to)
+}
+
+// AfterDeployment is the default post deployment behavior
+type AfterDeployment struct {
+	// tags allows setting image stream tags
+	tags client.ImageStreamTagsNamespacer
+}
+
+// Handle processes the post-deployment phase
+func (a *AfterDeployment) Handle(config *deployapi.DeploymentConfig, to *kapi.ReplicationController) error {
+	after := config.Spec.Strategy.After
+	if after == nil {
+		return nil
+	}
+
+	var errs []error
+	if len(after.TagImages) > 0 {
+		glog.Infof("Tagging images for successful deployment")
+		for _, action := range after.TagImages {
+			image, err := deployapi.TemplateImageForContainer(&to.Spec.Template.Spec, deployapi.DeploymentConfigHasTrigger(config), action.ContainerName)
```

Comment:
That's at the time the RC is created - not at the time it is processed by
the deployer.  If the RC is changed to *actually* use a different image,
that's what we want to use.  I.e. I want the thing that the majority of
pods used, as much as possible.

On Mon, Feb 1, 2016 at 10:48 AM, Michail Kargakis <notifications@github.com>
wrote:

> In pkg/cmd/infra/deployer/deployer.go
> <https://github.com/openshift/origin/pull/6925#discussion_r51434296>:
>
> > +	// tags allows setting image stream tags
> > +	tags client.ImageStreamTagsNamespacer
> > +}
> > +
> > +// Handle processes the post-deployment phase
> > +func (a *AfterDeployment) Handle(config *deployapi.DeploymentConfig, to *kapi.ReplicationController) error {
> > +	after := config.Spec.Strategy.After
> > +	if after == nil {
> > +		return nil
> > +	}
> > +
> > +	var errs []error
> > +	if len(after.TagImages) > 0 {
> > +		glog.Infof("Tagging images for successful deployment")
> > +		for _, action := range after.TagImages {
> > +			image, err := deployapi.TemplateImageForContainer(&to.Spec.Template.Spec, deployapi.DeploymentConfigHasTrigger(config), action.ContainerName)
>
> Well, the spec is a straight copy:
> https://github.com/openshift/origin/blob/733b7b76f455bf47cc617e4176afb29a14fdb601/pkg/deploy/util/util.go#L144
> +
> https://github.com/openshift/origin/blob/733b7b76f455bf47cc617e4176afb29a14fdb601/pkg/deploy/util/util.go#L205
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/6925/files#r51434296>.
>